pennylane_sf/gbs.py

```diff
@@ -0,0 +1,258 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Strawberry Fields variational GBS device
+========================================
+
+**Module name:** :mod:`pennylane_sf.gbs`
+
+.. currentmodule:: pennylane_sf.gbs
+
+The Strawberry Fields variational GBS device is a simulator that provides a way to encode
+variational parameters into GBS so that the gradient with respect to the output probability
+distribution is accessible.
+
+Classes
+-------
+
+.. autosummary::
+   StrawberryFieldsGBS
+
+Code details
+~~~~~~~~~~~~
+"""
+from collections import OrderedDict
+
+import numpy as np
+import pennylane as qml
+import strawberryfields as sf
+from pennylane.operation import Probability
+from pennylane.wires import Wires
+from strawberryfields.ops import GraphEmbed, MeasureFock
+from strawberryfields.utils import all_fock_probs_pnr
+from thewalrus.quantum import find_scaling_adjacency_matrix as rescale
+from thewalrus.quantum import photon_number_mean_vector
+
+from .expectations import identity
+from .simulator import StrawberryFieldsSimulator
+
+
+class StrawberryFieldsGBS(StrawberryFieldsSimulator):
+    r"""StrawberryFields variational GBS device for PennyLane.
+
+    This device provides a simulator that can embed variational parameters into GBS such that the
+    analytic gradient of the probability distribution is accessible.
```

Comment:
I've been thinking about this lately; probably good to have a link back to the device page, in case a user ends up here by accident:

```suggestion
    r"""StrawberryFields variational GBS device for PennyLane.

    This device provides a simulator that can embed variational parameters into GBS such that the
    analytic gradient of the probability distribution is accessible.
    
    For more details see :doc:`/devices/gbs`.
```